VMEncryption/main/check_util.py

```diff
@@ -326,3 +328,33 @@ def is_non_fatal_precheck_failure(self):
             detected = True
             self.logger.log("PRECHECK: Unsupported mount scheme detected")
         return detected
+
+    def preInitializationCheck(self,logger,imdsStoredResults,iMDSUtil):
+        '''This function is checking the VM compatibility for ADE'''
+        logger.log('Pre initialization check Start.')
+        try:
+            securityType = None
+            if imdsStoredResults.config_file_exists() and imdsStoredResults.get_security_type() != None:
+                securityType = imdsStoredResults.get_security_type()
+                logger.log("reading from imds stored results, security type is {0}.".format(securityType))
+            else:
+                securityType = iMDSUtil.get_vm_security_type()
+                #imds does not store security type for Standard or Basic type.
+                if(securityType=='' or securityType == None):
+                    securityType= CommonVariables.Standard
+                logger.log("reading from imds, security type is {0}.".format(securityType))
+            supported_Security_Types = [x for x in CommonVariables.supported_Security_Types if x.lower() == securityType.lower()]
```

Comment:
Could this be a bool using something like securityType.lower() in [x.lower() for x in CommonVariables.supported_Security_Types]?

Reply:
yes, agree.